xds/internal/balancer/clusterimpl/config.go

```diff
@@ -0,0 +1,83 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package clusterimpl
+
+import (
+	"encoding/json"
+
+	internalserviceconfig "google.golang.org/grpc/internal/serviceconfig"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+/*
+message XdsClusterImplLoadBalancingPolicyConfig {
```

Comment:
We cannot just use the proto message, because of the extra check we do on the sub balancer policies (`internalserviceconfig.BalancerConfig`).

Parent balancers shouldn't use child's policy struct directly (we do have some today, but we should change those).
Because the only validation we do is usually in `parseConfig`. Bypassing the JSON unmarshal also bypasses the validation, which can cause problems later.

Reply:
Would it be better to simply link to the serviceConfig proto message in grpc-proto repo rather than inlining it here in this comment? I'm only concerned about keeping this up-to-date if something changes in the actual proto.

Reply:
It's not in the proto. I deleted the comments.